features/expo/deviceInfo/hooks.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+import { enumToStr } from '../../../utils/enum';
+import * as Device from 'expo-device';
+import { DateTime } from 'luxon';
+
+import { DeviceInfo } from './constants';
+import { DeviceType } from './enums';
+
+export const useDeviceInfo = () => {
+    const [deviceInfo, setDeviceInfo] = useState<DeviceInfo>({
+        brand: null,
+        manufacturer: null,
+        deviceName: null,
+        modelId: null,
+        modelName: null,
+        osName: null,
+        osVersion: null,
+        osBuildId: null,
+        osInternalBuildId: null,
+        deviceType: null,
+        deviceYearClass: null,
+        isDevice: null,
+        supportedCpuArchitectures: null,
+        totalMemory: null,
+        uptime: null,
+        isJailBroken: null,
+    });
+
+    useEffect(() => {
+        const fetchDeviceInfo = () => {
+            Device.getDeviceTypeAsync().then(deviceType => {
+                setDeviceInfo(prev => ({...prev, deviceType: deviceType ? enumToStr(DeviceType, deviceType) : null}));
```

Comment:
If there is a null possibility of the `deviceType` variable, there is no need to update the state.
```suggestion
            Device.getDeviceTypeAsync().then(deviceType => {
                if (deviceType) {
                setDeviceInfo(prev => ({...prev, deviceType: enumToStr(DeviceType, deviceType)}));
                }
```